cli/analytics.ts

```diff
@@ -1,21 +1,40 @@
+import yargs from "yargs";
+
 import Analytics from "analytics-node";
-import { getConfigSettings, getConfigSettingsPath, upsertConfigSettings } from "df/cli/config";
-import { ynQuestion } from "df/cli/console";
-import { v4 as uuidv4 } from "uuid";
+import {getConfigSettings, getConfigSettingsPath, upsertConfigSettings} from "df/cli/config";
+import {ynQuestion} from "df/cli/console";
+import {INamedOption} from "df/cli/yargswrapper";
+import {v4 as uuidv4} from "uuid";
+
+export const noTrackOption: INamedOption<yargs.Options> = {
+  name: "no-track",
+  option: {
+    describe: `Disables analytics without asking the user. A settings.json file will be written at ${getConfigSettingsPath()} to store this setting`,
+    type: "boolean",
+    default: false
+  }
+};
 
 const analytics = new Analytics("eR24ln3MniE3TKZXkvAkOGkiSN02xXqw");
 
 let currentCommand: string;
+let allowAnonymousAnalytics: boolean;
 
-export async function maybeConfigureAnalytics() {
+export async function maybeConfigureAnalytics(noTrack: boolean) {
   const settings = await getConfigSettings();
   // We should only ask if users want to track analytics if they are in an interactive terminal;
```

Comment:
Nit: It feels like we should be reading the settings once rather than in every function, and overriding the tracking field (once) after it is read.

But this is a bit extra in the context of this PR.

Reply:
I have switched to Yet Another Global Variable